imap/mailbox.c

```diff
@@ -3693,6 +3693,13 @@ static int mailbox_update_carddav(struct mailbox *mailbox,
 #ifdef HAVE_LIBICALVCARD
         vcardcomponent *vcard = record_to_vcard_x(mailbox, new);
 
+        if (!vcard) {
+            syslog(LOG_ERR, "record_to_vcard_x failed for record %u:%s",
+                   cdata->dav.imap_uid, mailbox_name(mailbox));
```

Comment:
I see that this block was copied from the `#else` chunk below, which already used plain syslog.  But perhaps you could convert both to xsyslog too while you're in here anyway, something like:

```suggestion
            xsyslog(LOG_ERR, "record_to_vcard_x failed",
                             "record=<%u> mailbox=<%s>",
                             cdata->dav.imap_uid, mailbox_name(mailbox));
```

(n.b. I wrote this suggestion on the web, not in my editor.  I think it's right, but it might not be.)

Reply:
Fixed